### DISCUSSION TITLE
interpreting the recommended mutations in a one-by-one context

### DISCUSSION THREAD

**Discussion by avilella** [brianhie/efficient-evolution#15] [Q&A]
Hi,

I've been using this software for a while now, and I thought of a scheme for which I could attempt to have a "validated" or "rejected" tag for each of the somatic hypermutations of a monoclonal antibody, in the following way:

Given a monoclonal antibody observed in a single cell, I list each of the mutations with respect to the germline V-genes, and produce mAb protein sequences that are identical to the observed except for 1 aminoacid, in which the mutation has been reverted back to the germline (a.k.a 'step-1 mAb').

Then in parallel I then:
(1) submit this step-1 mAb to `efficient-evolution/bin/recommend.py`
(2) submit the mAb as it was observed the same way

Then I count how many time does the software mark the mutation as:
(1) 'rejected': when it suggests that the original mAb should mutate towards the step-1 mab.
(2) 'validated': when it suggests that the step-1 mAb should mutate towards the original mab.

Doing this for a collection of mAbs, I obtain many more 'rejected' mutations than I get 'validated', in about a 10 to 1 ratio, and I wonder if this is expected?
Should I put a limit to which mutations recommended by the software I take into account? E.g. by using the numbers in the output? Any recommendations?

**Comment by brianhie** [comment]
Interesting analysis, are you saying the approach recovers the somatic mutations about 10% of the time?

The underlying models are not explicitly trained to mature a germline antibody toward any target, so the fact that the algorithms can find somatic mutations when given an initial germline about 10% of the time (which seems high compared to random guessing, for example) is certainly interesting.

We limit mutations for experimental purposes using the number of recommending language models, which is output alongside the mutations. Though, this is a bit arbitrary and probably not correlated with experimental success.
